### PR TITLE
Add Conda package GitHub Action

### PR DESCRIPTION
**Pull request by hobu**
This PR adds a GitHub Action that builds the @conda-forge OSX, Windows, and Linux packages for every commit. In addition to testing a build on all three platforms, it makes the packages available as artifacts for individual download.

### CODE REVIEW

File: scripts/ci/conda/setup.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+cd proj.4-feedstock
+cat > recipe/recipe_clobber.yaml <<EOL
+source:
+  git_url: https://github.com/OSGeo/PROJ.git
+  git_rev: ${GITHUB_SHA}
+  url:
+  sha256:
+
+build:
+  number: 2112
```
Comment (rouault) [inline scripts/ci/conda/setup.sh:19]: why 2112 ?

Comment (hobu) [reply scripts/ci/conda/setup.sh:19]: > why 2112 ?

@warmerdam honoring 😄  More importantly, if someone is actually using these packages, we don't want them to clash with normally-installed ones from @conda-forge that would have build numbers incrementing from 0.

**Comment by hobu** [comment]
OK to merge? I've followed this up with another PR that implements automated generation of the plots and other enhancements to the doc builder that will use the conda package.

**Comment by rouault** [comment]
No objection from me. We'll probably need some mention in the doc to explain where/how to fetch those conda packages

**Pull request merged**